config/widgets/settings/generalOptionsTab.json

```diff
@@ -7,153 +9,105 @@
 			"image": "settingsWindow/lineHorizontal",
 			"rect": { "x" : 5, "y" : 289, "w": 365, "h": 3}
 		},
-		
 		{
-			"name": "settingGroupTitles",
-			"type": "labelGroup",
-			"font": "medium",
-			"alignment": "left",
-			"color": "yellow",
-			"items":
-			[
-				{
-					"position": {"x": 10, "y": 55},
-					"text": "vcmi.systemOptions.videoGroup"
-				},
-				{
-					"position": {"x": 380, "y": 55},
-					"text": "vcmi.systemOptions.audioGroup"
-				},
-				{
-					"position": {"x": 10, "y": 295},
-					"text": "vcmi.systemOptions.townsGroup"
-				}
-			]
-		},
-/////////////////////////////////////// Left section - Video Settings
-		{
-			"name": "resolutionLabel",
-			"type": "label",
-			"font": "medium",
-			"alignment": "left",
-			"color": "white",
-			"position": {"x": 45, "y": 85},
-			"text": "vcmi.systemOptions.resolutionButton.hover"
+			"type" : "labelTitle",
+			"position": {"x": 10, "y": 55},
+			"text": "vcmi.systemOptions.videoGroup"
 		},
 		{
-			"name": "resolutionButton",
-			"type": "button",
-			"position": {"x": 10, "y": 83},
-			"image": "settingsWindow/button32",
-			"help": "vcmi.systemOptions.resolutionButton",
-			"callback": "setGameResolution",
-			"items":
-			[
-				{
-					"name": "gearIcon",
-					"type": "picture",
-					"image": "settingsWindow/gear",
-					"position": {"x": 0, "y": 0 }
-				}
-			]
+			"type" : "labelTitle",
+			"position": {"x": 380, "y": 55},
+			"text": "vcmi.systemOptions.audioGroup"
 		},
-		
 		{
-			"name": "scalingLabel",
-			"type": "label",
-			"font": "medium",
-			"alignment": "left",
-			"color": "white",
-			"position": {"x": 45, "y": 115},
-			"text": "vcmi.systemOptions.scalingButton.hover"
+			"type" : "labelTitle",
+			"position": {"x": 10, "y": 295},
+			"text": "vcmi.systemOptions.townsGroup"
 		},
+/////////////////////////////////////// Left section - Video Settings
 		{
-			"name": "scalingButton",
-			"type": "button",
-			"position": {"x": 10, "y": 113},
-			"image": "settingsWindow/button32",
-			"help": "vcmi.systemOptions.scalingButton",
-			"callback": "setGameScaling",
-			"items":
-			[
+			"type" : "verticalLayout",
+			"customType" : "labelDescription",
+			"position" : {"x": 45, "y": 85},
+			"items" : [
 				{
-					"name": "gearIcon",
-					"type": "picture",
-					"image": "settingsWindow/gear",
-					"position": {"x": 0, "y": 0 }
+					"name": "resolutionLabel",
+					"text": "vcmi.systemOptions.resolutionButton.hover",
+					"created" : "desktop"
+				},
+				{
+					"name": "scalingLabel",
+					"text": "vcmi.systemOptions.scalingButton.hover"
+				},
+				{
+					"text": "vcmi.systemOptions.fullscreenBorderless.hover",
+					"created" : "desktop"
+				},
+				{
+					"text": "vcmi.systemOptions.fullscreenExclusive.hover",
+					"created" : "desktop"
+				},
+				{
+					"text": "vcmi.systemOptions.framerateButton.hover"
+				},
+				{
+					"text": "core.genrltxt.577"
```

Comment:
@krs0 
What are you asking for is handled by "layouts" like this one.
For example this is definition of 6 labels, that:
- share same settings, as defined by "labelDescription" custom type (left-aligned, medium font, white color)
- placed vertically, at 30px intervals - defined by "verticalLayout" custom type
- (TODO) some labels only appear on desktop systems, so you won't have "fullscreen" option on Android and it won't result in any empty space between unavailable options - since only actually created labels will be added to layout.

Note that only container itself has position defined (45,85). Positions for individual labels are automatically generated based on layout settings

Reply:
In other words, if you want to add one more label all you have to do is add one more section like
```json
{
	"text": "vcmi.systemOptions.myNewOption.hover"
},
```
And all positioning will be handled automatically